api/handlers/changes.js

```diff
@@ -208,25 +205,33 @@ var getChanges = function(feed) {
     },
     (err, responses) => {
       endTimer(`getChanges().requests:${chunks.length}`, startTime);
-
-      if (feed.res.finished) {
-        // Don't write to the response if it has already ended. The change
-        // will be picked up in the subsequent changes request.
-        return;
-      }
-      cleanUp(feed);
       if (err) {
-        feed.res.write(error(503, 'Error processing your changes'));
-      } else {
+        return errorResponse(feed);
+      }
+
+      // if relevant ids have changed, update validated ids, getChanges again
+      const originalValidatedIds = feed.validatedIds;
+      bindServerIds(feed, err => {
```

Comment:
This is the new bit - if your `validatedIds` have changed then we need to go back and request the changes again from the same seq using the updated list of IDs.